src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -1628,6 +1628,12 @@ private void ReportResultsToSubmission(BuildResult result)
 
                     _overallBuildSuccess = _overallBuildSuccess && (_buildSubmissions[result.SubmissionId].BuildResult.OverallResult == BuildResultCode.Success);
 
+                    if (submission.BuildRequestData != null && submission.BuildRequestData.Flags.HasFlag(BuildRequestDataFlags.ClearProjectRootElementCacheAfterBuild))
+                    {
+                        // Reset the project root element cache if specified which ensures that projects will be re-loaded from disk
+                        _buildParameters?.ProjectRootElementCache?.Clear();
```

Comment:
Wonder if it would be better to also clear the ConfigCache and ResultsCache at this point (similar to what BuildManager.ResetCaches is doing). Anything held in the caches after restore completes is just memory waste.